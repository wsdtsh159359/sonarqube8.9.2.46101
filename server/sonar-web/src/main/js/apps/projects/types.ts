/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { ComponentQualifier } from '../../types/component';

export interface Project {
  analysisDate?: string;
  isFavorite?: boolean;
  key: string;
  leakPeriodDate?: string;
  measures: T.Dict<string>;
  name: string;
  projects?: number;
  qualifier: ComponentQualifier;
  tags: string[];
  visibility: T.Visibility;
  needIssueSync?: boolean;
}

export interface Facet {
  [value: string]: number;
}

export interface Facets {
  [property: string]: Facet;
}
