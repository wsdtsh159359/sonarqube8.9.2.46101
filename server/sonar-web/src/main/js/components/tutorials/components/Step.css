/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
.onboarding-step {
  position: relative;
  padding-left: 34px;
  margin-bottom: var(--gridSize);
}

.onboarding-step.no-step-number {
  padding-left: 0;
}

.onboarding-step:not(.is-open):not(.is-finished) {
  opacity: 0.4;
}

.onboarding-step .boxed-group-actions {
  height: var(--controlHeight);
  line-height: var(--controlHeight);
}

.onboarding-step hr {
  margin-left: -54px;
}

.onboarding-step-number {
  position: absolute;
  top: 15px;
  left: 15px;
  width: 24px;
  height: 24px;
  line-height: 24px;
  border-radius: 24px;
  background-color: #b9b9b9;
  color: #fff;
  font-size: var(--mediumFontSize);
  text-align: center;
}

.onboarding-step.is-open .onboarding-step-number {
  background-color: var(--darkBlue);
}

.onboarding-step.is-finished {
  cursor: pointer;
  outline: none;
}

.onboarding-step ol.list-styled {
  list-style: none;
  padding-left: 0;
  counter-reset: step-counter;
}

.onboarding-step .markdown {
  line-height: inherit;
}

.onboarding-step ul.list-styled li {
  margin-top: var(--gridSize);
  margin-bottom: var(--gridSize);
}

.onboarding-step ol.list-styled > li {
  position: relative;
  counter-increment: li;
  margin-bottom: calc(2 * var(--gridSize));
}

.onboarding-step ol.list-styled > li::before {
  display: inline-block;
  width: 16px;
  height: 16px;
  margin-right: 8px;
  color: #fff;
  font-size: 12px;
  line-height: 16px;
  direction: rtl;
  text-align: center;
  background-color: #4b9fd5;
  border-radius: 50%;
  content: counter(li);
}
