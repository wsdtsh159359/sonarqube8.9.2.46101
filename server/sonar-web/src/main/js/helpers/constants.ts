/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { colors } from '../app/theme';
import { AlmKeys } from '../types/alm-settings';
import { ComponentQualifier } from '../types/component';
import { IssueScope, IssueType } from '../types/issues';

export const SEVERITIES = ['BLOCKER', 'CRITICAL', 'MAJOR', 'MINOR', 'INFO'];
export const STATUSES = ['OPEN', 'REOPENED', 'CONFIRMED', 'RESOLVED', 'CLOSED'];
export const ISSUE_TYPES: T.IssueType[] = [
  IssueType.Bug,
  IssueType.Vulnerability,
  IssueType.CodeSmell,
  IssueType.SecurityHotspot
];
export const SOURCE_SCOPES = [
  { scope: IssueScope.Main, qualifier: ComponentQualifier.File },
  { scope: IssueScope.Test, qualifier: ComponentQualifier.TestFile }
];
export const RULE_TYPES: T.RuleType[] = ['BUG', 'VULNERABILITY', 'CODE_SMELL', 'SECURITY_HOTSPOT'];
export const RULE_STATUSES = ['READY', 'BETA', 'DEPRECATED'];

export const CHART_COLORS_RANGE_PERCENT = [
  colors.green,
  colors.lightGreen,
  colors.yellow,
  colors.orange,
  colors.red
];

export const CHART_REVERSED_COLORS_RANGE_PERCENT = [
  colors.red,
  colors.orange,
  colors.yellow,
  colors.lightGreen,
  colors.green
];

export const RATING_COLORS = [
  colors.green,
  colors.lightGreen,
  colors.yellow,
  colors.orange,
  colors.red
];

export const PROJECT_KEY_MAX_LEN = 400;

export const ALM_DOCUMENTATION_PATHS = {
  [AlmKeys.Azure]: '/documentation/analysis/azuredevops-integration/',
  [AlmKeys.BitbucketServer]: '/documentation/analysis/bitbucket-integration/',
  [AlmKeys.BitbucketCloud]: '/documentation/analysis/bitbucket-cloud-integration/',
  [AlmKeys.GitHub]: '/documentation/analysis/github-integration/',
  [AlmKeys.GitLab]: '/documentation/analysis/gitlab-integration/'
};

export const IMPORT_COMPATIBLE_ALMS = [
  AlmKeys.Azure,
  AlmKeys.BitbucketServer,
  AlmKeys.GitHub,
  AlmKeys.GitLab
];
