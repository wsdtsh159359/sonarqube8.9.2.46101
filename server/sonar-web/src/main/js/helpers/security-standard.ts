/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { Standards } from '../types/security';

export function getStandards(): Promise<Standards> {
  return import('./standards.json').then(x => x.default);
}

export function renderCWECategory(standards: Standards, category: string): string {
  const record = standards.cwe[category];
  if (!record) {
    return `CWE-${category}`;
  } else if (category === 'unknown') {
    return record.title;
  } else {
    return `CWE-${category} - ${record.title}`;
  }
}

export function renderOwaspTop10Category(
  standards: Standards,
  category: string,
  withPrefix = false
): string {
  const record = standards.owaspTop10[category];
  if (!record) {
    return addPrefix(category.toUpperCase(), 'OWASP', withPrefix);
  } else {
    return addPrefix(`${category.toUpperCase()} - ${record.title}`, 'OWASP', withPrefix);
  }
}

export function renderSansTop25Category(
  standards: Standards,
  category: string,
  withPrefix = false
): string {
  const record = standards.sansTop25[category];
  return addPrefix(record ? record.title : category, 'SANS', withPrefix);
}

export function renderSonarSourceSecurityCategory(
  standards: Standards,
  category: string,
  withPrefix = false
): string {
  const record = standards.sonarsourceSecurity[category];
  if (!record) {
    return addPrefix(category.toUpperCase(), 'SONAR', withPrefix);
  } else if (category === 'others') {
    return record.title;
  } else {
    return addPrefix(record.title, 'SONAR', withPrefix);
  }
}

function addPrefix(title: string, prefix: string, withPrefix: boolean) {
  return withPrefix ? `${prefix} ${title}` : title;
}
