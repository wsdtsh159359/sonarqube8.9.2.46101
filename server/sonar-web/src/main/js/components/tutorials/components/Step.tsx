/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
/* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/no-noninteractive-tabindex */
import * as classNames from 'classnames';
import * as React from 'react';
import './Step.css';

interface Props {
  finished?: boolean;
  onOpen?: VoidFunction;
  open: boolean;
  renderForm: () => React.ReactNode;
  renderResult?: () => React.ReactNode;
  stepNumber?: number;
  stepTitle: React.ReactNode;
}

export default function Step(props: Props) {
  const { finished, open, stepNumber, stepTitle } = props;
  const className = classNames('boxed-group', 'onboarding-step', {
    'is-open': open,
    'is-finished': finished,
    'no-step-number': stepNumber === undefined
  });

  const clickable = !open && finished && props.onOpen !== undefined;

  const handleClick = (event: React.MouseEvent<HTMLDivElement>) => {
    event.preventDefault();
    if (props.onOpen !== undefined) {
      props.onOpen();
    }
  };

  return (
    <div
      className={className}
      onClick={clickable ? handleClick : undefined}
      role={clickable ? 'button' : undefined}
      tabIndex={clickable ? 0 : undefined}>
      {stepNumber !== undefined && <div className="onboarding-step-number">{stepNumber}</div>}
      {!open && props.renderResult && props.renderResult()}
      <div className="boxed-group-header">
        <h2>{stepTitle}</h2>
      </div>
      {!open && <div className="boxed-group-inner" />}
      <div className={classNames({ hidden: !open })}>{props.renderForm()}</div>
    </div>
  );
}
