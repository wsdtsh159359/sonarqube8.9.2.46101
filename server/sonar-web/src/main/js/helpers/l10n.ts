/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { toNotSoISOString } from 'sonar-ui-common/helpers/dates';
import SonarUiCommonInitializer, { DEFAULT_LOCALE } from 'sonar-ui-common/helpers/init';
import {
  get as loadFromLocalStorage,
  save as saveInLocalStorage
} from 'sonar-ui-common/helpers/storage';
import { fetchL10nBundle } from '../api/l10n';
import { L10nBundle, L10nBundleRequestParams } from '../types/l10n';

const L10N_BUNDLE_LS_KEY = 'l10n.bundle';

export async function loadL10nBundle() {
  const bundle = await getLatestL10nBundle().catch(() => ({
    locale: DEFAULT_LOCALE,
    messages: {}
  }));

  SonarUiCommonInitializer.setLocale(bundle.locale).setMessages(bundle.messages);
  // No need to load english (default) bundle, it's coming with react-intl
  if (bundle.locale !== DEFAULT_LOCALE) {
    const [intlBundle, intl] = await Promise.all([
      import(`react-intl/locale-data/${bundle.locale}`),
      import('react-intl')
    ]);

    intl.addLocaleData(intlBundle.default);
  }

  return bundle;
}

export async function getLatestL10nBundle() {
  const browserLocale = getPreferredLanguage();
  const cachedBundle = loadL10nBundleFromLocalStorage();

  const params: L10nBundleRequestParams = {};

  if (browserLocale) {
    params.locale = browserLocale;

    if (
      cachedBundle.locale &&
      browserLocale.startsWith(cachedBundle.locale) &&
      cachedBundle.timestamp &&
      cachedBundle.messages
    ) {
      params.ts = cachedBundle.timestamp;
    }
  }

  const { effectiveLocale, messages } = await fetchL10nBundle(params).catch(response => {
    if (response && response.status === 304) {
      return {
        effectiveLocale: cachedBundle.locale || browserLocale || DEFAULT_LOCALE,
        messages: cachedBundle.messages ?? {}
      };
    } else {
      throw new Error(`Unexpected status code: ${response.status}`);
    }
  });

  const bundle = {
    timestamp: toNotSoISOString(new Date()),
    locale: effectiveLocale,
    messages
  };

  saveL10nBundleToLocalStorage(bundle);

  return bundle;
}

export function getCurrentL10nBundle() {
  return loadL10nBundleFromLocalStorage();
}

function getPreferredLanguage() {
  return window.navigator.languages ? window.navigator.languages[0] : window.navigator.language;
}

function loadL10nBundleFromLocalStorage() {
  let bundle: L10nBundle;

  try {
    bundle = JSON.parse(loadFromLocalStorage(L10N_BUNDLE_LS_KEY) ?? '{}');
  } catch {
    bundle = {};
  }

  return bundle;
}

function saveL10nBundleToLocalStorage(bundle: L10nBundle) {
  saveInLocalStorage(L10N_BUNDLE_LS_KEY, JSON.stringify(bundle));
}
