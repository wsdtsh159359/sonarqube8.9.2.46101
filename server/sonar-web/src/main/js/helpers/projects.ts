/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { ProjectKeyValidationResult } from '../types/component';
import { PROJECT_KEY_MAX_LEN } from './constants';

export function validateProjectKey(projectKey: string): ProjectKeyValidationResult {
  // This is the regex used on the backend:
  //   [\p{Alnum}\-_.:]*[\p{Alpha}\-_.:]+[\p{Alnum}\-_.:]*
  // See sonar-core/src/main/java/org/sonar/core/component/ComponentKeys.java
  const regex = /^[\w\-.:]*[a-z\-_.:]+[\w\-.:]*$/i;
  if (projectKey.length === 0) {
    return ProjectKeyValidationResult.Empty;
  } else if (projectKey.length > PROJECT_KEY_MAX_LEN) {
    return ProjectKeyValidationResult.TooLong;
  } else if (regex.test(projectKey)) {
    return ProjectKeyValidationResult.Valid;
  } else {
    return /^[0-9]+$/.test(projectKey)
      ? ProjectKeyValidationResult.OnlyDigits
      : ProjectKeyValidationResult.InvalidChar;
  }
}
