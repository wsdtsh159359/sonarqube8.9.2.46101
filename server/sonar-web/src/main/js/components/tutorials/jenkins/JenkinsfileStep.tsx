/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { translate } from 'sonar-ui-common/helpers/l10n';
import AllSet from '../components/AllSet';
import RenderOptions from '../components/RenderOptions';
import Step from '../components/Step';
import { BuildTools } from '../types';
import DotNet from './buildtool-steps/DotNet';
import Gradle from './buildtool-steps/Gradle';
import Maven from './buildtool-steps/Maven';
import Other from './buildtool-steps/Other';

export interface JenkinsfileStepProps {
  component: T.Component;
  open: boolean;
}

// To remove when CFamily is includ in this tutorial
type BuildToolsWithoutCFamily = Exclude<BuildTools, BuildTools.CFamily>;

const BUILDTOOL_COMPONENT_MAP: {
  [x in BuildToolsWithoutCFamily]: React.ComponentType<{ component: T.Component }>;
} = {
  [BuildTools.Maven]: Maven,
  [BuildTools.Gradle]: Gradle,
  [BuildTools.DotNet]: DotNet,
  [BuildTools.Other]: Other
};

export default function JenkinsfileStep(props: JenkinsfileStepProps) {
  const { component, open } = props;
  const [buildTool, setBuildTool] = React.useState<BuildToolsWithoutCFamily | undefined>(undefined);
  return (
    <Step
      finished={false}
      open={open}
      renderForm={() => (
        <div className="boxed-group-inner">
          <ol className="list-styled">
            <li>
              {translate('onboarding.build')}
              <RenderOptions
                checked={buildTool}
                name="buildtool"
                onCheck={value => setBuildTool(value as BuildToolsWithoutCFamily)}
                optionLabelKey="onboarding.build"
                options={Object.keys(BUILDTOOL_COMPONENT_MAP)}
              />
            </li>
            {buildTool !== undefined &&
              React.createElement(BUILDTOOL_COMPONENT_MAP[buildTool], { component })}
          </ol>
          {buildTool !== undefined && (
            <>
              <hr className="huge-spacer-top huge-spacer-bottom" />
              <AllSet />
            </>
          )}
        </div>
      )}
      stepNumber={3}
      stepTitle={translate('onboarding.tutorial.with.jenkins.jenkinsfile.title')}
    />
  );
}
