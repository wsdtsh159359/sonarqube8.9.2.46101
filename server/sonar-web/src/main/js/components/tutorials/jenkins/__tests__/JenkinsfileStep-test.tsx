/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import { mockComponent } from '../../../../helpers/testMocks';
import RenderOptions from '../../components/RenderOptions';
import Step from '../../components/Step';
import { BuildTools } from '../../types';
import JenkinsfileStep, { JenkinsfileStepProps } from '../JenkinsfileStep';
import { renderStepContent } from '../test-utils';

it('should render correctly', () => {
  const wrapper = shallowRender();
  expect(wrapper).toMatchSnapshot('Step wrapper');
  expect(renderStepContent(wrapper)).toMatchSnapshot('initial content');
});

it('should render correctly for Maven', () => {
  const wrapper = shallowRender();
  selectBuildTool(wrapper, BuildTools.Maven);
  expect(
    wrapper
      .find(Step)
      .props()
      .renderForm()
  ).toMatchSnapshot();
});

it('should render correctly for Gradle', () => {
  const wrapper = shallowRender();
  selectBuildTool(wrapper, BuildTools.Gradle);
  expect(renderStepContent(wrapper)).toMatchSnapshot();
});

it('should render correctly for .NET', () => {
  const wrapper = shallowRender();
  selectBuildTool(wrapper, BuildTools.DotNet);
  expect(renderStepContent(wrapper)).toMatchSnapshot();
});

it('should render correctly for Other', () => {
  const wrapper = shallowRender();
  selectBuildTool(wrapper, BuildTools.Other);
  expect(renderStepContent(wrapper)).toMatchSnapshot();
});

function selectBuildTool(wrapper: ShallowWrapper<JenkinsfileStepProps>, tool: BuildTools) {
  const content = new ShallowWrapper(renderStepContent(wrapper) as JSX.Element);
  content.find(RenderOptions).prop('onCheck')(tool);
}

function shallowRender(props: Partial<JenkinsfileStepProps> = {}) {
  return shallow<JenkinsfileStepProps>(
    <JenkinsfileStep component={mockComponent()} open={true} {...props} />
  );
}
