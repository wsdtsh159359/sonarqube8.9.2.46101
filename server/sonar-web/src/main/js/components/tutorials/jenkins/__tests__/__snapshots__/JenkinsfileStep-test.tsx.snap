// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly for .NET 1`] = `
<div
  className="boxed-group-inner"
>
  <ol
    className="list-styled"
  >
    <li>
      onboarding.build
      <RenderOptions
        checked="dotnet"
        name="buildtool"
        onCheck={[Function]}
        optionLabelKey="onboarding.build"
        options={
          Array [
            "maven",
            "gradle",
            "dotnet",
            "other",
          ]
        }
      />
    </li>
    <DotNet
      component={
        Object {
          "breadcrumbs": Array [],
          "key": "my-project",
          "name": "MyProject",
          "qualifier": "TRK",
          "qualityGate": Object {
            "isDefault": true,
            "key": "30",
            "name": "Sonar way",
          },
          "qualityProfiles": Array [
            Object {
              "deleted": false,
              "key": "my-qp",
              "language": "ts",
              "name": "Sonar way",
            },
          ],
          "tags": Array [],
        }
      }
    />
  </ol>
  <React.Fragment>
    <hr
      className="huge-spacer-top huge-spacer-bottom"
    />
    <Connect(withAppState(AllSet)) />
  </React.Fragment>
</div>
`;

exports[`should render correctly for Gradle 1`] = `
<div
  className="boxed-group-inner"
>
  <ol
    className="list-styled"
  >
    <li>
      onboarding.build
      <RenderOptions
        checked="gradle"
        name="buildtool"
        onCheck={[Function]}
        optionLabelKey="onboarding.build"
        options={
          Array [
            "maven",
            "gradle",
            "dotnet",
            "other",
          ]
        }
      />
    </li>
    <Gradle
      component={
        Object {
          "breadcrumbs": Array [],
          "key": "my-project",
          "name": "MyProject",
          "qualifier": "TRK",
          "qualityGate": Object {
            "isDefault": true,
            "key": "30",
            "name": "Sonar way",
          },
          "qualityProfiles": Array [
            Object {
              "deleted": false,
              "key": "my-qp",
              "language": "ts",
              "name": "Sonar way",
            },
          ],
          "tags": Array [],
        }
      }
    />
  </ol>
  <React.Fragment>
    <hr
      className="huge-spacer-top huge-spacer-bottom"
    />
    <Connect(withAppState(AllSet)) />
  </React.Fragment>
</div>
`;

exports[`should render correctly for Maven 1`] = `
<div
  className="boxed-group-inner"
>
  <ol
    className="list-styled"
  >
    <li>
      onboarding.build
      <RenderOptions
        checked="maven"
        name="buildtool"
        onCheck={[Function]}
        optionLabelKey="onboarding.build"
        options={
          Array [
            "maven",
            "gradle",
            "dotnet",
            "other",
          ]
        }
      />
    </li>
    <Maven
      component={
        Object {
          "breadcrumbs": Array [],
          "key": "my-project",
          "name": "MyProject",
          "qualifier": "TRK",
          "qualityGate": Object {
            "isDefault": true,
            "key": "30",
            "name": "Sonar way",
          },
          "qualityProfiles": Array [
            Object {
              "deleted": false,
              "key": "my-qp",
              "language": "ts",
              "name": "Sonar way",
            },
          ],
          "tags": Array [],
        }
      }
    />
  </ol>
  <React.Fragment>
    <hr
      className="huge-spacer-top huge-spacer-bottom"
    />
    <Connect(withAppState(AllSet)) />
  </React.Fragment>
</div>
`;

exports[`should render correctly for Other 1`] = `
<div
  className="boxed-group-inner"
>
  <ol
    className="list-styled"
  >
    <li>
      onboarding.build
      <RenderOptions
        checked="other"
        name="buildtool"
        onCheck={[Function]}
        optionLabelKey="onboarding.build"
        options={
          Array [
            "maven",
            "gradle",
            "dotnet",
            "other",
          ]
        }
      />
    </li>
    <Other
      component={
        Object {
          "breadcrumbs": Array [],
          "key": "my-project",
          "name": "MyProject",
          "qualifier": "TRK",
          "qualityGate": Object {
            "isDefault": true,
            "key": "30",
            "name": "Sonar way",
          },
          "qualityProfiles": Array [
            Object {
              "deleted": false,
              "key": "my-qp",
              "language": "ts",
              "name": "Sonar way",
            },
          ],
          "tags": Array [],
        }
      }
    />
  </ol>
  <React.Fragment>
    <hr
      className="huge-spacer-top huge-spacer-bottom"
    />
    <Connect(withAppState(AllSet)) />
  </React.Fragment>
</div>
`;

exports[`should render correctly: Step wrapper 1`] = `
<Step
  finished={false}
  open={true}
  renderForm={[Function]}
  stepNumber={3}
  stepTitle="onboarding.tutorial.with.jenkins.jenkinsfile.title"
/>
`;

exports[`should render correctly: initial content 1`] = `
<div
  className="boxed-group-inner"
>
  <ol
    className="list-styled"
  >
    <li>
      onboarding.build
      <RenderOptions
        name="buildtool"
        onCheck={[Function]}
        optionLabelKey="onboarding.build"
        options={
          Array [
            "maven",
            "gradle",
            "dotnet",
            "other",
          ]
        }
      />
    </li>
  </ol>
</div>
`;
